sunpy/net/helioviewer.py

```diff
@@ -109,10 +117,15 @@ def get_closest_image(self, date, observatory=None, instrument=None,
         >>> print(metadata['date'])  # doctest: +REMOTE_DATA
         2012-01-01T00:00:07.000
         """
+        data_dictionary = self.data_sources
```

Comment:
Why is this defined here?

Reply:
We can define this in the init function. We need the dictionary because it is used to get the value from the partial key. We can define it for once and then use.

Reply:
But it is used one?

Reply:
Well you use it once, so why not just call the original and not create a new variable?